src/background/services/keyAutoAdd.ts

```diff
@@ -178,25 +178,49 @@ export class KeyAutoAddService {
       return false;
     }
   }
+
+  static async registerContentScripts({ browser }: Pick<Cradle, 'browser'>) {
```

Comment:
I would make this a non-static method, since it is already registered in the container, to avoid passing around the `browser` instance. We will have to add the service in the background if we do this.